consoles/ssh_screen.pm

```diff
@@ -52,4 +59,41 @@ sub do_read
     }
     return undef;
 }
+
+sub type_string
+{
+    my ($self, $nargs) = @_;
+
+    bmwqemu::log_call(%$nargs);
+
+    my $text           = $nargs->{text};
+    my $terminate_with = $nargs->{terminate_with} // '';
+    my $written        = 0;
+
+    $text .= "\cC" if ($terminate_with eq 'ETX');
+
+    while ($written < length($text)) {
+        my $chunk = $self->ssh_channel->write(substr($text, $written));
+
+        if (!defined($chunk)) {
+            my ($errcode, $errname, $errstr) = $self->ssh_connection->error;
+
+            croak "Lost SSH connection to SUT: $errstr"
+              if $errcode != LIBSSH2_ERROR_EAGAIN;
+            select(undef, undef, undef, 0.1);
+        } elsif ($chunk < 0) {
+            # Old Net::SSH2 error signaling
+            croak "Lost SSH connection to SUT"
```

Comment:
Are we actually connecting to the SUT here? This console seems to be rather generic and also from the PR title I would assume that it is intended for generic use. Note that at least when using the svrit backend there seems to be more work to do for connecting to the actual SUT (and not just the host running the VM), e.g. calling `virsh console` (see `start_serial_grab` in `svirt.pm`).

Reply:
Yes, this serial terminal is intended to connect directly to the SUT, unlike the existing sshVirtshSUT console which connects to the hypervisor instead and attaches to a virtual serial device.

This terminal is generic in the sense that it'll work with any backend as long as the SUT runs SSH daemon on a port accessible from the worker.

Here's an experimental run with generic SSH terminal on ZKVM: https://openqa.suse.de/tests/5175184

Reply:
Ah, ok - you're simply starting an SSH daemon within the SUT. That makes sense.